ddtrace/contrib/starlette/patch.py

```diff
@@ -65,3 +53,40 @@ def unpatch():
     setattr(starlette, "_datadog_patch", False)
 
     _u(starlette.applications.Starlette, "__init__")
+
+    # We need to check that Fastapi instrumentation hasn't already unpatched these
+    if isinstance(starlette.routing.Route.handle, ObjectProxy):
+        _u(starlette.routing.Route, "handle")
+
+    if isinstance(starlette.routing.Mount.handle, ObjectProxy):
+        _u(starlette.routing.Mount, "handle")
+
+
+def traced_handler(wrapped, instance, args, kwargs):
+    # Since handle can be called multiple times for one request, we take the path of each instance
+    # Then combine them at the end to get the correct resource names
+    scope = get_argument_value(args, kwargs, 0, "scope")
+
+    # Add the path to the resource_paths list
+    if "resource_paths" not in scope["datadog"]:
+        scope["datadog"]["resource_paths"] = [instance.path]
+    else:
+        scope["datadog"]["resource_paths"].append(instance.path)
+    import pdb
+
+    pdb.set_trace()
+    if len(scope["datadog"].get("request_spans")) == len(scope["datadog"]["resource_paths"]):
```

Comment:
🤔 

it is probably going to be cleaner, if you pull out these values ahead of time rather than continuing to call `.get()` which is "slow"

```python
request_spans = scope["datadog"].get("request_spans", [])  # type: List[Span]
resource_paths = scope["datadog"]["resource_paths"]  # type: List[str]

if len(request_spans) == len(resource_paths):
    for index, span in enumerate(request_spans):
        ...
```

Reply:
I cannot....

Reply:
yeah, I am not sure either.

but I do wonder if we should have a fallback for the case when we have at least one request span but X number of resource paths.

e.g. if we had:

```
request_spans = [Span()]
resource_paths = ["/sub-app", "/some/route/{name}"]
```

We might still want to allow this to work, e.g.

```python
if len(request_spans) == len(resource_paths):
    for index, span in enumerate(request_spans):
        ...
elif request_spans and resource_paths:
    # at least always update the root asgi span resource name
    request_spans[0].resource = "".join(resource_paths)
else:
    ... log a warning? is this actually a warning or normal behavior

```

Reply:
wdyt? overkill?

Reply:
I think that seems reasonable